src/main/java/io/debezium/connector/jdbc/SinkRecordDescriptor.java

```diff
@@ -177,7 +177,8 @@ private FieldDescriptor(Schema schema, String name, boolean key, DatabaseDialect
             this.typeName = type.getTypeName(dialect, schema, key);
 
             this.name = name;
-            this.columnName = SchemaUtils.getSourceColumnName(schema).orElse(name);
+            // Avoid avro schema reuse of Geometry record structure to get same column name
+            this.columnName = isStructType() ? name : SchemaUtils.getSourceColumnName(schema).orElse(name);
```

Comment:
@harveyyue Should this be limited to struct only? IIUC any type that has metadata in parameters like Decimal will be affected.

Reply:
> @harveyyue Should this be limited to struct only? IIUC any type that has metadata in parameters like Decimal will be affected.

Yes, only struct type, primitive type is work fine.

Reply:
@harveyyue The problem as descibed above is that if there are two arbitrary types differing with schema parameters only then only one (probably the first) is used and the rest is forgotten. In that case I believe the schema parameters for all types should be used. Also IMHO the problem with schema paramaters is thus not limited to this place but might be present elsewhere too. WDYT?

Reply:
> @harveyyue The problem as descibed above is that if there are two arbitrary types differing with schema parameters only then only one (probably the first) is used and the rest is forgotten. In that case I believe the schema parameters for all types should be used. Also IMHO the problem with schema paramaters is thus not limited to this place but might be present elsewhere too. WDYT?

Yes, I think this case is limited when using avro, protobuf or other format converter scenario.

Reply:
@harveyyue I merged the SMT PR. To make sure the issue is really addresed could you please prepare an integration test running for MySQL with Apicurio and Avro that will have two geometry columns and two decimal columns with different constraints and check that everything is properly proagated?
That way we'll be 100 % sure that the general issue was nailed.
Thanks